File: 02-09-2025_Find_the_Number_of_Ways_to_Place_People_I.cpp
class Solution {
public:
    int numberOfPairs(vector<vector<int>>& points) {
        sort(points.begin(), points.end(), [](const vector<int>& a, const vector<int>& b) {
            if (a[0] == b[0]) return a[1] > b[1];
            return a[0] < b[0];
        });

        int count = 0;
        int n = points.size();
        for (int i = 0; i < n; i++) {
            int t = points[i][1];
            int b = -1;
            for (int j = i + 1; j < n; j++) {
                if (b < points[j][1] && points[j][1] <= t) {
                    count++;
                    b = points[j][1];
                    if (t == b) break;
                }
            }
        }
        return count;
    }
};

File: 02-09-2025_Find_the_Number_of_Ways_to_Place_People_I.py
class Solution:
    def numberOfPairs(self, points: List[List[int]]) -> int:
        points.sort(key= lambda x: (x[0],-x[1]))
        count=0
        for i in range(0,len(points)):
            t=points[i][1]
            b=-1
            for j in range(i+1,len(points)):
                if (b<points[j][1] and points[j][1]<=t):
                    count+=1
                    b=points[j][1]
                    if t==b:
                        break
        return count
